### PR TITLE
Document creation

### PR DESCRIPTION
**Pull request by honourfish**
## Overview
* Added new Document sample based on @henry739's original work

## Testing
* Tested the new document sample, also tested it doesn't recreate
* Regression tested by running WIPP sample, and checking it doesn't recreate

<sub> Mention [stepsize] in a comment if you'd like to report some technical debt. See examples [here](https://app.stepsize.com/api/demo-pr-redirect).</sub>

### CODE REVIEW

File: archivist_samples/document/document.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+        return self.arch.events.create(
+            self.asset["identity"],
+            props=props,
+            attrs=attrs,
+            asset_attrs=asset_attrs,
+            confirm=True,
+        )
+
+    # Withdraw version of the document
+    def withdraw(self, document: dict, version: str, hash: str, name: str):
```
Comment (eccles) [inline archivist_samples/document/document.py:156]: hash is a builtin and this will fail checks

**Review by eccles — CHANGES_REQUESTED** [review]

File: archivist_samples/document/document.py (hunk 1 above)
Comment (honourfish) [reply archivist_samples/document/document.py]: renamed to doc_hash!

File: archivist_samples/document/document.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+import logging
+
+from copy import copy
+from typing import Dict, List, Optional
+
+# pylint:disable=unused-import      # To prevent cyclical import errors forward referencing is used
+# pylint:disable=cyclic-import      # but pylint doesn't understand this feature
+
+from archivist import archivist as type_helper
```
Comment (Serhiy1) [inline archivist_samples/document/document.py:15]: After doing type-hinting for RF, I've come down to a particular style.  

```python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
     from archivist.archivisti mport Archivist

```
Then use the type hints in quotation marks,  

```python
arch: "Archivist",
```
Additionally, you no longer need a majority of the typing imports. 
```python
Optional[list] -> "list|None"
Union[list, str] -> "list|str"
List[str] -> "list[str]"
```

Comment (honourfish) [reply archivist_samples/document/document.py]: Excellent summary! have amended document.py to follow this pattern for type hinting.

**Review by JAG-UK — APPROVED** [review]
Reviewed in browser - Looks great :-)

**Review by eccles — APPROVED** [review]

**Pull request merged**